fragments/infra-boot.sh

```diff
@@ -1,86 +1,253 @@
 
-    HOME=/root
-    mv /etc/systemd/system/multi-user.target.wants/docker-storage-setup.service $HOME
+# check if this is an Atomic host
+function is_atomic_host() {
+    [ -e /run/ostree-booted ]
+}
+
+# remove the docker storage setup service link and re-load the systemd config
+function systemd_docker_disable_storage_setup() {
+    mv /etc/systemd/system/multi-user.target.wants/docker-storage-setup.service /root
     systemctl daemon-reload
+}
 
-    # Setup Docker Storage Volume Group
-    if ! [ -b /dev/vdb ]; then
-      echo "ERROR: device /dev/vdb does not exist" >&2
-      notify_failure "device /dev/vdb does not exist"
+# All hosts must have an external disk device (cinder?) for docker storage
+function docker_check_for_storage_device() {
+    # DOCKER_VOLUME_DEVICE=$1
+    if [ -b $1 ]
+    then
+        notify_failure \
+            "docker volume device $1 does not exist"
     fi
+}
 
-    systemctl enable lvm2-lvmetad
-    systemctl start lvm2-lvmetad
-cat << EOF > /etc/sysconfig/docker-storage-setup
-DEVS=/dev/vdb
+# configure docker storage
+function docker_set_storage_device() {
+    # DOCKER_VOLUME_DEVICE=$1
+    cat << EOF > /etc/sysconfig/docker-storage-setup
+DEVS=$1
 VG=docker-vg
 EOF
-    /usr/bin/docker-storage-setup || notify_failure "failed to run docker-storage-setup"
-    systemctl start docker --ignore-dependencies || notify_failure "failed to start docker"
+}
 
-    docker pull jprovaznik/ooshift-dns
-    docker run -d -p 53:53/udp -v /var/log:/var/log -v /etc/hosts:/etc/hosts -v /etc/dnsmasq.conf:/etc/dnsmasq.conf -v /etc/resolv.conf.local:/etc/resolv.conf --name dnsmasq jprovaznik/ooshift-dns || notify_failure "failed to run dns docker image"
-    docker pull jprovaznik/ooshift-heat-agent
-    docker run -d --privileged --ipc=host --net=host --pid=host -e HOST=/host -e NAME=rhel-tools -v /run:/run -v /var/log:/var/log -v /etc/localtime:/etc/localtime -v ~/.ssh:/root/.ssh -v /:/host -v /etc/ansible:/etc/ansible -v /var/lib/heat-cfntools:/var/lib/heat-cfntools --name heat-agent jprovaznik/ooshift-heat-agent || notify_failure "failed to run heat-agent docker image"
-else
-    systemctl is-enabled os-collect-config || notify_failure "os-collect-config service is not installed or enabled"
-    if [ "$SKIP_DNS" != "true" ]; then
-        yum install -y dnsmasq || notify_failure "can't install dnsmasq"
-        systemctl enable dnsmasq || notify_failure "can't enable dnsmasq"
-        systemctl restart dnsmasq || notify_failure "can't start dnsmasq"
-    fi
 
-    yum install -y git httpd-tools || notify_failure "could not install httpd-tools"
+#
+# --- OpenShift Auxiliary Service Containers
+#
+
+# The DNS container runs Dnsmasq from root file space
+function start_dns_container() {
+    # DNS_CONTAINER_IMAGE=$1
+    docker pull $1 ||
+        notify_failure "failed to pull dns docker image: $1"
+    docker run \
+           --name dnsmasq \
+           --daemon \
```

Comment:
'--deamon' seems to be an invalid parameter:
Mar 22 14:33:11 localhost cloud-init: flag provided but not defined: --daemon
Mar 22 14:33:11 localhost cloud-init: See 'docker run --help'.